### PR TITLE
Handle Detached Segments in MergeTree Client getPositions

### LINKED ISSUE #3753: MergeTree.getPosition returns 0 even if segment doesn't exist.
## Describe the bug
While using mergeTree.getPosition(segment) to retrieve the position of a segment, I realized that even if my segment doesn't exist on the DDS or even if I pass in as a parameter something that is not a segment, the method returns 0 as the position. This is unexpected, given that I would expect to have a -1 returned for something that is not part of the DDS's mergetree

### To Reproduce
1. create sharedsTring.
2. use sharedString.getPosition('123')
3. THe method above will return 0 even if '123' is not a segment, making it confusing for developers to know if '123' is actually in the mergetree or not.

### Expected behavior
-1 returned as a position of an invalid entry parameter or non-existing segment

### PR DESCRIPTION
**Pull request by anthony-murphy**
fixes #3753

### CODE REVIEW

File: packages/dds/sequence/src/sequence.ts [hunk 1]
```diff
@@ -243,7 +243,12 @@ export abstract class SharedSegmentSequence<T extends MergeTree.ISegment>
         return this.client.getLength();
     }
 
-    public getPosition(segment): number {
+    /**
```
Comment (vladsud) [inline packages/dds/sequence/src/sequence.ts:246]: You can likely use @inheritDoc format here to refer to client API

**Review by vladsud — APPROVED** [review]

File: packages/dds/sequence/src/sequence.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/dds/sequence/src/sequence.ts:246]: I don't think @inheritdoc  works with composition

File: packages/dds/merge-tree/src/test/client.getPostion.spec.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    it("Deleted Segment",()=>{
+        client.removeRangeLocal(segPos, segPos + 1);
+        assert.notStrictEqual(segment.removedSeq, undefined);
+        const pos = client.getPosition(segment);
+        assert.strictEqual(pos, segPos);
+    });
+
+    it("Detached Segment",()=>{
+        client.applyMsg(client.makeOpMessage(client.removeRangeLocal(segPos, segPos + 1), 1));
+        // so some work and move the client's min seq forward, so zamboni runs
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/test/client.getPostion.spec.ts:43]:
```suggestion
        // do some work and move the client's min seq forward, so zamboni runs
```

File: packages/dds/sequence/src/sequence.ts (hunk 1 above)
Comment (vladsud) [reply packages/dds/sequence/src/sequence.ts:246]: As far as I can see, it does not care. You just point to any symbol

**Pull request merged**